source/portable/NetworkInterface/STM32/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+if (NOT ( (FREERTOS_PLUS_TCP_NETWORK_IF STREQUAL "STM32") ) )
+    return()
+endif()
+
+#------------------------------------------------------------------------------
+add_library( freertos_plus_tcp_network_if STATIC )
+
+target_sources( freertos_plus_tcp_network_if
+  PRIVATE
+    NetworkInterface.c
+)
```

Comment:
Could we integrate the driver sources with the cmake module? For example, we could add specific driver sources based on the value of a cmake cache variable. 

This is my proposition:
```suggestion
set( FREERTOS_PLUS_TCP_STM32_IF_DRIVER "None" CACHE STRING "The driver sources to use with STM32 Network interface" )

target_sources( freertos_plus_tcp_network_if
  PRIVATE
    NetworkInterface.c
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},F4>:
      Drivers/F4/stm32f4xx_hal_eth.c>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},F7>:
      Drivers/F7/stm32f7xx_hal_eth.c>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},H5>:
      Drivers/H5/stm32h5xx_hal_eth_ex.c>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},H5>:
      Drivers/H5/stm32h5xx_hal_eth.c>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},H7>:
      Drivers/H7/stm32h7xx_hal_eth_ex.c>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},H7>:
      Drivers/H7/stm32h7xx_hal_eth.c>
)

target_include_directories( freertos_plus_tcp_network_if 
  PUBLIC
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},F4>:
      Drivers/F4>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},F7>:
      Drivers/F7>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},H5>:
      Drivers/H5>
    $<$<STREQUAL:${FREERTOS_PLUS_TCP_STM32_IF_DRIVER},H7>:
      Drivers/H7>
)
```